### DISCUSSION TITLE
Cant install additional content / unclear documentation

### DISCUSSION THREAD

**Discussion by AFCMS** [Neverball/neverball#261] [Q&A] [answered]
I downloaded some levels from neverforum a while ago, and tried to install them following the process described here:

https://github.com/Neverball/neverball/blob/95a04d1b9796cb65a365d3a0d241c9876dff68a0/doc/manual.txt#L79-#L82

So I put the ZIP files in my `.neverball`:
![image](https://user-images.githubusercontent.com/61794590/134006148-48774f2d-6ab6-4444-95be-ee170c521b47.png)

But the game don't load anything :(

System: Ubuntu 20.04

Neverball Version:
![image](https://user-images.githubusercontent.com/61794590/134006519-06ae40df-849f-4eee-ac31-f71c1f351cca.png)

**Comment by brunetton** [comment]
Same here; did you find a solution ?

**Comment by parasti** [comment]
Two conditions have to be met:

1. The content must be packaged correctly: a set-...txt file must be located at the top level of the archive. ZIP files with a single top-level folder will not work.
2. Neverball 1.6 must be compiled with PhysicsFS support.

Without seeing the exact ZIP file structure, it's impossible to say why it doesn't work.

**Reply by brunetton** [reply]
Thanks a lot for your answer.

I forgot to mention:
- I'm speaking of never**putt** here, **not** neverball
- I'm sure I'm using the correct data directory (~/.neverball/), as log file is updated when I run Neverputt

> Two conditions have to be met:
> 
> 1. The content must be packaged correctly: a set-...txt file must be located at the top level of the archive. ZIP files with a single top-level folder will not work.

I'll assume that this file must be named `holes-...txt`, not `set-...txt`, as this is the case in every Neverputt custom levels sets.

> 2. Neverball 1.6 must be compiled with PhysicsFS support.

I've no idea if this is the case. How to check ? I'm using Ubuntu 21.10 (impish) version

> Without seeing the exact ZIP file structure, it's impossible to say why it doesn't work.

Here are the steps to reproduce:
- download `cliche.zip` from [this page](http://neverforum.com/fmpbo/levels.php) (direct link: http://neverforum.com/fmpbo/addons/levels/putt/cliche.zip)
- the zip structure looks fine to me:
  - `holes-cliche.txt` at the root of archive, referencing sol files (terrain files I guess) in `map-cliche/`subdir
- copy `cliche.zip` to `~/.neverball/` (I checked rights, I can read the file, the process; that belongs to me, should be able to read the file too)
- (re)start Neverputt
- no new entry menu in courses list :(

## Workaround

As @AFCMS suggested, a workaround is to extract archive directly to `~/.neverball/`directory (so `~/.neverball/holes-cliche.txt` exists). 

This works for **one** custom levels pack; but when trying to add another one ([CSY Puttworks Gamma](http://neverforum.com/fmpbo/addons/levels/putt/puttworksgamma.zip) for example) there are conflicts: it seems that a lot (all ?) of packs uses the `shot-putt` directory name to put preview image. Of course I can rename each one, edit all the txt files and change the first lines to match the new directory name ... but this is a lot of work and it's not simple ! Same problem for `back`directories used to store background texture (I guess).

**Comment by AFCMS** [comment, marked as answer]
I solved my problem by just extracting the archives (I forgot to update the discussion)
